src/rfc3986/builder.py

```diff
@@ -61,7 +73,7 @@ def __init__(
         self.scheme = scheme
         self.userinfo = userinfo
         self.host = host
-        self.port = port
+        self.port = str(port) if port is not None else port
```

Comment:
Hm, Why are we converting this to a string here? The builder is really our own thing that doesn't exist in other libraries to the best of my knowledge, so we can do this, but just trying to understand why

Reply:
Ah, I see we call `normalize_authority` with it and that is really expecting a string. Huh. Must not have good test coverage here. :(

Reply:
I did only this to be consistent with the `URIBuilder.add_port` method, which converts inputs to a string and stores that, and is shown in the docs as accepting integers. iirc, the docs say that directly instantiating the builder isn't recommended, but I figured it wouldn't hurt to have parity type-wise on the off chance it *does* get instantiated directly, and also so that the constructor and the builder method have more similar behavior overall.